pandas/core/indexes/base.py

```diff
@@ -500,7 +500,9 @@ def __new__(
             result = RangeIndex(start=data, copy=copy, name=name)
             if dtype is not None:
                 return result.astype(dtype, copy=False)
-            return result
+            # error: Incompatible return value type (got "MultiIndex",
+            # expected "Self")
+            return result  # type: ignore[return-value]
```

Comment:
I'm not sure why mypy didn't find this previously: `self == other` can return an `ExtensionArray` and it seems that it doesn't define `__invert__`